addons/hr_timesheet/security/hr_timesheet_security.xml

```diff
@@ -91,5 +91,18 @@
             <field name="model_id" ref="model_timesheets_analysis_report"/>
             <field name="domain_force">[('company_id', 'in', company_ids)]</field>
         </record>
+
+        <record id="timesheet_analysis_report_user" model="ir.rule">
+            <field name="name">Timesheets Analysis Report user</field>
+            <field name="model_id" ref="model_timesheets_analysis_report"/>
+            <field name="domain_force">[('user_id', '=', user.id)]</field>
+            <field name="groups" eval="[(4, ref('group_hr_timesheet_user'))]"/>
+        </record>
+
+        <record id="timesheet_analysis_report_approver" model="ir.rule">
+            <field name="name">Timesheets Analysis Report approver</field>
+            <field name="model_id" ref="model_timesheets_analysis_report"/>
+            <field name="groups" eval="[(4, ref('group_hr_timesheet_approver'))]"/>
+        </record>
```

Comment:
It is weird for me, the user will continue to see more records with the report. 🤔 
I am wondering if the rules should not be the same than the ones defined for `account.analytic.line` model at the same file? 🤔 

I agree the `('project_id', '!=', False)` tuple is not necessary for those rules of the report since we only fetch the timesheets in that report. 🙂

Reply:
As discussed, a user should not have more information in the report (`timesheets.analysis.report` model) than they have access to for the `account.analytic.line` model. :)